### PR TITLE
Fixed problem that the oauth keys are loaded on boot

### PR DESCRIPTION
**Pull request by ResuBaka**
That should not be allowed as then you can't run passport:install. So
with that fix it is possible. You only have one problem when you change
the load path of the oauth keys after you have loaded the ServiceProvider
of this package.

I have fixed it for the default way passport is used, eg you don't change the location of the oauth keys.

### CODE REVIEW

File: src/Providers/LighthouseGraphQLPassportServiceProvider.php [hunk 1]
```diff
@@ -23,6 +23,16 @@ class LighthouseGraphQLPassportServiceProvider extends ServiceProvider
      */
     public function boot()
     {
+        [$publicKey, $privateKey] = [
+            Passport::keyPath('oauth-public.key'),
+            Passport::keyPath('oauth-private.key'),
+        ];
+
+        /** Don't try to boot when you can't load the passport keys. */
+        if (!file_exists($publicKey) || !file_exists($privateKey)) {
```
Comment (joselfonseca) [inline src/Providers/LighthouseGraphQLPassportServiceProvider.php]: What if the user has the Public and private keys as Strings in the env vars? https://laravel.com/docs/6.x/passport#deploying-passport I think using file_exists may be an issue if we are using that other method.

Comment (ResuBaka) [reply src/Providers/LighthouseGraphQLPassportServiceProvider.php]: I have changed it so it is not needed anymore.

**Comment by ResuBaka** [comment]
@joselfonseca 

The problem was that your workaround does not work in one case. Which is when you clone a project and then you don't have a app `APP_KEY` which is needed for creating the passport keys but because you can't create them you can't create and `APP_KEY` locally. 

We have found a workaround for it with just remove this package create both keys and then reinstall it, but that was a bit to much of a workaround for me/us.

But with how i know changed it it should just work out of the box and it extend the server the same way as Passport does it.

I have not tested if the social login works with that fix as have no project where I have it configured so I could test it.
Maybe you have a project where you could test this PR or when you don't have one I could try to set one up and test it if it still is working.

**Comment by joselfonseca** [comment]
@ResuBaka thanks, looks a lot better now.

**Comment by ResuBaka** [comment]
No Problem :smiley:. 

Do you create a new release for that fix or do you wait a bit for it? @joselfonseca

**Comment by ResuBaka** [comment]
I just saw that you created a release. I should just have waited a bit :(.

**Comment by joselfonseca** [comment]
👍

**Pull request merged**